ifrs17/Report/ReportMutableScopesInteractive.ipynb

```diff
@@ -0,0 +1,580 @@
+                "\n//     {",
+                "\n//         //this.scopes = scopes; ",
+                "\n//         this.report = report;",
+                "\n//         this.interactiveObject = interactiveObject;",
+                "\n//         storage = new ReportStorage(workspace, report);",
+                "\n//     }",
+                "\n     ",
+                "\n//     public Systemorph.Vertex.InteractiveObjects.InteractiveObjectView PvReportFormsEntity =>",
+                "\n//         interactiveObject.CreateView(nameof(PvReportFormsEntity),",
+                "\n//             state => state.GetScope<PvReport>(nameof(PvReport), o => o.WithStorage(storage)));",
+                "\n    ",
+                "\n//     public Systemorph.Vertex.InteractiveObjects.InteractiveObjectView PvReportGrid =>",
+                "\n//         interactiveObject.CreateView(nameof(PvReportGrid),",
+                "\n//             state => {",
+                "\n//                 var scope = state.GetScope<PvReport>(nameof(PvReport), o => o.WithStorage(storage));",
+                "\n//                 var filters = scope.GetFilters(); // Not used and should be improved",
+                "\n//                 return scope.ToReportAsync();",
+                "\n//             });",
+                "\n",
+                "\n",
+                "\n//     //public IIfrs17Report PresentValues => reportUniverse.GetScope<PvReport>();",
+                "\n//     // public IIfrs17Report RiskAdjustments => reportUniverse.GetScope<RaReport>();",
+                "\n//     // public IIfrs17Report WrittenActuals => reportUniverse.GetScope<WrittenReport>();",
+                "\n//     // public IIfrs17Report AccrualActuals => reportUniverse.GetScope<AccrualReport>();",
+                "\n//     // public IIfrs17Report DeferralActuals => reportUniverse.GetScope<DeferralReport>();",
+                "\n//     // public IIfrs17Report FulfillmentCashflows => reportUniverse.GetScope<FcfReport>();",
+                "\n//     // public IIfrs17Report ExperienceAdjustments => reportUniverse.GetScope<ExpAdjReport>();",
+                "\n//     // public IIfrs17Report TechnicalMargins => reportUniverse.GetScope<TmReport>();",
+                "\n//     // public IIfrs17Report AllocatedTechnicalMargins => reportUniverse.GetScope<CsmReport>();",
+                "\n//     // public IIfrs17Report ActuarialLrc => reportUniverse.GetScope<ActLrcReport>();",
+                "\n//     // public IIfrs17Report Lrc => reportUniverse.GetScope<LrcReport>();",
+                "\n//     // public IIfrs17Report ActuarialLic => reportUniverse.GetScope<ActLicReport>();",
+                "\n//     // public IIfrs17Report Lic => reportUniverse.GetScope<LicReport>();",
+                "\n//     // public IIfrs17Report FinancialPerformance => reportUniverse.GetScope<FpReport>();",
```

Comment:
If each report is a separate method of the Ifrs17Intercative class, can we make each report virtual? If the requirements in particular project are different from the default, it would help a lot. I discussed this briefly with @dcolleoni today, and he also agrees with this approach.

Reply:
No, this should not happen and this is legacy code to be removed.